### DISCUSSION TITLE
Maui Blazor Hybrid but for Uno WASM?

### DISCUSSION THREAD

**Discussion by hawkerm** [unoplatform/uno#14845] [Q&A] [answered]
Is there anything like [.NET Maui Blazor Hybrid](https://learn.microsoft.com/en-us/aspnet/core/blazor/hybrid/tutorials/maui) but for wrapping an Uno WASM app instead?

i.e. being able to leverage a native container with a WebView around an Uno WASM app, but then can leverage the direct connection to .NET that being in that shell provides the WebView for better WASM/interop performance?

**Comment by jeromelaban** [comment, marked as answer]
Great question! There are multiple technical limitations with this approach, but the biggest one is that Uno is tightly and synchronously integrated with the DOM events (loaded-like events, specifically). As a result, operations performed in the DOM from uno must be synchronous as well, and the browser integrations do not prove synchronous APIs anymore. 

If Uno started to use async APIs, the result would be lots of visual glitches while the UI is being built, because the browser would render partial visual tree elements. Blazor is able to do this because the rendering is effectively asynchronous and its engine does not need communication in a synchronous fashion.
